File: src/UnwindMC.Tests/Helpers/TempDirectory.cs
using System;
using System.IO;

namespace UnwindMC.Tests.Helpers
{
    class TempDirectory : IDisposable
    {
        public TempDirectory(string dirPath)
        {
            Path = dirPath;
            Directory.CreateDirectory(Path);
        }
        public string Path { get; }

        public void Dispose()
        {
            Directory.Delete(Path, recursive: true);
        }
    }
}

File: src/UnwindMC.Tests/Util/TempDirectory.fs
module TempDirectory

open System
open System.IO

type T(dirPath: string) =
    let path: string = dirPath

    do
        Directory.CreateDirectory(path) |> ignore

    member self.Path: string = path
    
    interface IDisposable with
        member self.Dispose(): unit =
            Directory.Delete(path, true)
